src/uid.cpp

```diff
@@ -0,0 +1,4 @@
+#include "uid.h"
+
+template <>
+uid_factory<UID_ITEM> *uid_factory<UID_ITEM>::instance = nullptr;
```

Comment:
To implement this as a factory `uid::deserialize( JsonIn &js )` needs to know where the factory is located and there isn't any other way to provide this